dg/lib/openshift/data/global_pull_secret_data.py

```diff
@@ -0,0 +1,199 @@
+            be printed
+        """
+
+        if use_json:
+            click.echo(json.dumps(self._secret_get_response, indent="\t", sort_keys=True))
+        else:
+            auth_list: List[List[str]] = []
+            auths = self._secret_get_response["auths"]
+
+            for image_registry, auth in auths.items():
+                search_result = regex.match("(.*)\\:(.*)", base64.standard_b64decode(auth["auth"]).decode("utf-8"))
+
+                if search_result is None:
+                    raise DataGateCLIException("…")
+
+                image_registry_password = search_result.group(2)
+
+                if len(image_registry_password) > 5:
+                    image_registry_password = image_registry_password[0:5] + "…"
+
+                auth_list_element: List[str] = [image_registry, search_result.group(1), image_registry_password]
+
+                auth_list.append(auth_list_element)
+
+            auth_list.sort(key=lambda auth_list_element: auth_list_element[0])
+            click.echo(tabulate(auth_list, headers=["registry location", "username", "password"]))
+
+    def get_credentials(self, registry_location: str) -> Optional[Credentials]:
+        """Returns credentials for the given registry location
+
+        Parameters
+        ----------
+        registry_location
+            registry location for which credentials shall be returned
+
+        Returns
+        -------
+        str
+            credentials for the given registry location
+        """
+
+        search_result: Any = jmespath.search(f'auths."{registry_location}".auth', self._secret_get_response)
+
+        return self._decode_credentials(search_result) if search_result is not None else None
+
+    def get_json_patch(self) -> Any:
+        """Returns the global pull secret as a JSON Patch object
+
+        Notes
+        -----
+        http://jsonpatch.com
+
+        Returns
+        -------
+        Any
+            global pull secret as a JSON Patch object
+        """
+
+        return [
+            {
+                "op": "replace",
+                "path": "/data/.dockerconfigjson",
+                "value": base64.standard_b64encode(json.dumps(self._secret_get_response).encode()).decode("utf-8"),
+            }
+        ]
+
+    def set_credentials(self, registry_location: str, username: str, password: str):
+        """Sets credentials for the given registry location
+
+        Parameters
+        ----------
+        registry_location
+            registry location for which credentials shall be set
+        username
+            registry location username
+        password
+            registry location password
+        """
+
+        auth: AuthDict = {"auth": base64.standard_b64encode(f"{username}:{password}".encode()).decode("utf-8")}
+
+        auths = self._secret_get_response["auths"]
+        auths[registry_location] = auth
+
+        self._secret_get_response["auths"] = dict(sorted(auths.items(), key=lambda auth: auth[0]))
```

Comment:
As I'm not entirely sure how the json content looks like being extracted above, please make sure the complete document is in a flattened state and not pretty printed or anything (Otherwise the patching process probably won't work)

Reply:
This logic was well tested and works like a charm.